### PR TITLE
Update docs with new usage tracking info

### PR DESCRIPTION
**Pull request by NishchayKarle**
# Description

Update usage tracking docs with information about the tracking levels and dashboard

## Type of change

- Update to human readable text: Documentation/error messages/comments

### CODE REVIEW

**Review by benclifford — APPROVED** [review]

File: docs/userguide/usage_tracking.rst [hunk 1]
```diff
@@ -1,82 +1,170 @@
...
+* **Level 2:** Configuration information including provider, executor, and launcher names.
+* **Level 3:** Workflow execution details, including the number of applications run, failures, and execution time.
 
-We have chosen opt-in collection rather than opt-out with the hope that developers and researchers
-will choose to send us this information. The reason is that we need this data - it is a requirement for funding.
+By enabling usage tracking, you support Parsl's development. 
 
-By opting-in, and allowing these statistics to be reported back, you are explicitly supporting the
-further development of Parsl.
+**To opt-in, set ``usage_tracking`` to the desired level (1, 2, or 3) in the configuration object (`parsl.config.Config`).**
```
Comment (benclifford) [inline docs/userguide/usage_tracking.rst:30]: the backquote formatting doesn't work when i build this locally, inside the ** bold block. you could remove the bold block.

File: docs/userguide/usage_tracking.rst [hunk 2]
```diff
@@ -1,82 +1,170 @@
...
+Parsl uses an **Opt-in** model for usage tracking, allowing users to decide if they wish to participate. Usage statistics are crucial for improving software reliability and help focus development and maintenance efforts on the most used components of Parsl. The collected data is used solely for enhancements and reporting and is not shared in its raw form outside of the Parsl team.
 
 Why are we doing this?
 ----------------------
 
-The Parsl development team receives support from government funding agencies. For the team to continue to
-receive such funding, and for the agencies themselves to argue for funding, both the team and the agencies
-must be able to demonstrate that the scientific community is benefiting from these investments. To this end,
-it is important that we provide aggregate usage data about such things as the following:
+The Parsl development team relies on funding from government agencies. To sustain this funding and advocate for continued support, it is essential to show that the scientific community benefits from these investments.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:11]:
```suggestion
The Parsl development team relies on funding from government agencies. To sustain this funding and advocate for continued support, it is essential to show that the research community benefits from these investments.
```

File: docs/userguide/usage_tracking.rst [hunk 3]
```diff
@@ -1,82 +1,170 @@
...
-* Parsl exit codes
+Choose the data you share with Usage Tracking Levels.
 
-By participating in this project, you help justify continuing support for the software on which you rely.
-(see :ref:`What is sent? <what-is-sent>` below).
+**Usage Tracking Levels:**
 
-Opt-In
-------
+* **Level 1:** Basic information such as Python version, Parsl version, and platform details.
```
[the remark this answers is not in the document]
Comment (danielskatz) [reply docs/userguide/usage_tracking.rst:24]: We might also want to say what "platform details" means

File: docs/userguide/usage_tracking.rst [hunk 4]
```diff
@@ -1,82 +1,170 @@
...
+* **Level 2:** Configuration information including provider, executor, and launcher names.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:25]:
```suggestion
* **Level 2:** Level 1 information and configuration information including provider, executor, and launcher names.
```

File: docs/userguide/usage_tracking.rst [hunk 5]
```diff
@@ -1,82 +1,170 @@
...
+* **Level 3:** Workflow execution details, including the number of applications run, failures, and execution time.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:26]:
```suggestion
* **Level 3:** Level 2 information and workflow execution details, including the number of applications run, failures, and execution time.
```

File: docs/userguide/usage_tracking.rst [hunk 6]
```diff
@@ -1,82 +1,170 @@
...
-* Run UUID
-* Start and end times
-* Number of executors used
-* Number of failures
-* Parsl and Python version
-* OS and OS version
-
+The data collected depends on the tracking level selected:
+
+* **Level 1:** Basic information such as Python version, Parsl version, and platform details.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:52]:
```suggestion
* **Level 1:** Only basic information such as Python version, Parsl version, and platform details.
```

Comment (danielskatz) [reply docs/userguide/usage_tracking.rst:52]: We might want to say what "platform details" includes

File: docs/userguide/usage_tracking.rst [hunk 7]
```diff
@@ -1,82 +1,170 @@
...
+* **Level 2:** Configuration information including provider, executor, and launcher names.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:53]:
```suggestion
* **Level 2:** Level 1 information and configuration information including provider, executor, and launcher names.
```

File: docs/userguide/usage_tracking.rst [hunk 8]
```diff
@@ -1,82 +1,170 @@
...
+* **Level 3:** Workflow execution details, including the number of applications run, failures, and execution time.
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:54]:
```suggestion
* **Level 3:** Level 2 information and workflow execution details, including the number of applications run, failures, and execution time.
```

File: docs/userguide/usage_tracking.rst [hunk 9]
```diff
@@ -1,82 +1,170 @@
...
+Data is sent twice per run:
 
+1. At the start of the script.
+2. Upon script completion (for Tracking Level 3).
 
 What will the data be used for?
 -------------------------------
 
-The data will be used for reporting purposes to answer questions such as:
+The data will help answer key questions, including:
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:122]:
```suggestion
The data will help the Parsl team understand Parsl usage and make development and maintenance decisions, including:
```

**Comment by danielskatz** [comment]
looks good overall; I've suggested some wording changes

File: docs/userguide/usage_tracking.rst [hunk 10]
```diff
@@ -21,13 +21,13 @@ Choose the data you share with Usage Tracking Levels.
 
 **Usage Tracking Levels:**
 
-* **Level 1:** Basic information such as Python version, Parsl version, and platform details.
-* **Level 2:** Configuration information including provider, executor, and launcher names.
-* **Level 3:** Workflow execution details, including the number of applications run, failures, and execution time.
+* **Level 1:** Only basic information such as Python version, Parsl version, and platform details (Linux, MacOS etc.).
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:24]: Is platform details just OS? If so, say so...  This is still too vague to me

Comment (NishchayKarle) [reply docs/userguide/usage_tracking.rst:24]: Yes. It's just the OS name. I'll make the change

File: docs/userguide/usage_tracking.rst [hunk 11]
```diff
@@ -21,7 +21,7 @@ Choose the data you share with Usage Tracking Levels.
 
 **Usage Tracking Levels:**
 
-* **Level 1:** Only basic information such as Python version, Parsl version, and platform details (Linux, MacOS etc.).
+* **Level 1:** Only basic information such as Python version, Parsl version, and platform name (Linux, MacOS etc.).
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:24]:
```suggestion
* **Level 1:** Only basic information such as Python version, Parsl version, and platform name (Linux, MacOS, etc.)
```

File: docs/userguide/usage_tracking.rst [hunk 12]
```diff
@@ -49,7 +49,7 @@ What is sent?
 
 The data collected depends on the tracking level selected:
 
-* **Level 1:** Only basic information such as Python version, Parsl version, and platform details (Linux, MacOS etc.).
+* **Level 1:** Only basic information such as Python version, Parsl version, and platform name (Linux, MacOS etc.).
```
Comment (danielskatz) [inline docs/userguide/usage_tracking.rst:52]:
```suggestion
* **Level 1:** Only basic information such as Python version, Parsl version, and platform name (Linux, MacOS, etc.)
```

**Review by benclifford — APPROVED** [review]

**Pull request merged**